File: api-java/src/main/java/fr/huiitre/tools/modules/core/feedback/application/usecase/GetAllFeedbacksUseCase.java
package fr.huiitre.tools.modules.core.feedback.application.usecase;

import java.util.List;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import fr.huiitre.tools.modules.core.feedback.application.port.FeedbackRepository;
import fr.huiitre.tools.modules.core.feedback.infrastructure.FeedbackEntity;

@Service
public class GetAllFeedbacksUseCase implements SecuredUseCase {

    private final FeedbackRepository feedbackRepository;

    public GetAllFeedbacksUseCase(FeedbackRepository feedbackRepository) {
        this.feedbackRepository = feedbackRepository;
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.ADMIN;
    }

    public List<FeedbackEntity> execute() {
        return feedbackRepository.findAllSortedByDateDesc();
    }
}

File: api/Modules/Core/Feedback/Application/Usecases/GetAllFeedbacksUseCase.cs
using Tools.Api.Modules.Core.Feedback.Application.Dto;
using Tools.Api.Modules.Core.Feedback.Application.Ports;
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;
using Tools.Api.Modules.Core.Security.Domain;

namespace Tools.Api.Modules.Core.Feedback.Application.Usecases;

public sealed class GetAllFeedbacksUseCase(
    IFeedbackRepository feedbackRepository,
    UseCaseAuthorizer authoriser
) : SecuredUseCase(authoriser)
{
    protected override RoleCode RequiredRole => RoleCode.Admin;

    public async Task<List<FeedbackDto>> Execute()
    {
        return await feedbackRepository.FindAllSortedByDateDesc();
    }
}